### ISSUE TITLE
Rename completely ignores conflicts in collection initializers

### ISSUE THREAD

**Issue by SLaks** [dotnet/roslyn#3524]
Source:

```C#
class C : IEnumerable {
	void Add<T>(T x) { }
	[Obsolete]
	void M(int x) { }
	public IEnumerator GetEnumerator() => new C() { 1 }.GetEnumerator();
}
```

Renaming `M` to `Add` will show no conflicts in the rename, but will show an obsolete warning in the initializer (you silently changed which method it binds to)

**Comment by SLaks** [comment]
Shorter repro:

```C#
class C : List<int> {
	[Obsolete]
	void M(int x) => new C { 2 };
}
```

(doesn't warn due to #3525)

**Comment by SLaks** [comment]
It also fails to report removing the initializer entirely:


```C#
class C : IEnumerable {
	void Add(int x) { }
	public IEnumerator GetEnumerator() => new C() { 1 }.GetEnumerator();
}
```

Renaming `Add` to anything else should show a conflict. (or, better yet, resolve to an explicit call)

**Comment by dpoeschl** [comment]
Related to internal work item #658058

**Comment by Pilchie** [comment]
Moving to unknown, until we know whether there is enough desire to invest in this sort of confilct detection before adding new refactorings.
